core/renderers/common/path_object.ts

```diff
@@ -268,37 +268,33 @@ export class PathObject implements IPathObject {
     connectionPath: string,
     offset: Coordinate,
     rtl: boolean,
-  ) {
-    if (this.connectionHighlights.has(connection)) {
-      if (this.currentHighlightMatchesNew(connection, connectionPath, offset)) {
-        return;
-      }
-      this.removeConnectionHighlight(connection);
+  ): SVGElement {
+    const transformation =
+      `translate(${offset.x}, ${offset.y})` + (rtl ? ' scale(-1 1)' : '');
+
+    const previousHighlight = this.connectionHighlights.get(connection);
+    if (previousHighlight) {
+      // Since a connection already exists, make sure that its path and
+      // transform are correct.
+      previousHighlight.setAttribute('d', connectionPath);
+      previousHighlight.setAttribute('transform', transformation);
```

Comment:
FYI @rachel-fenichel this bit confuses me. The connection seems to still magically update even if these aren't re-set across block movements or changes (despite not recreating the SVG). Either I'm doing something completely wrong here, or completely misunderstanding something because that doesn't sound correct to me.

Reply:
To be extra clear (since I realized now upon re-reading that this could've been worded better): if I remove these lines, the highlighting still seems to work and I'm having difficulty reasoning through the dataflow sufficiently well as to understand why that makes sense. I understand that the block paths basically get recreated upon movements and such, and that in turn means that the connection highlights are recomputed, but my other changes in this PR should prevent this block from being called if the path already exists. Unless the `Connection` objects themselves are being recreated (which I highly doubt since it would break focus), the original highlight transform shouldn't be updating. :)

Reply:
If you remove which lines?

Reply:
Continued in https://github.com/google/blockly/pull/8928#discussion_r2069354218. No clue why a new thread started.

Reply:
Resolving since this is continued in the other thread.